### PR TITLE
[Snappi]: Adding support for breakout links

### PR DESCRIPTION
**Pull request by selldinesh**
### Description of PR

Summary:Adding support for breakout links
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?
This PR supports the addition of breakout ports 
#### How did you verify/test it?
Instead of mentioning Card info , we can dirently mention the fanout port info in the ansible/links file example Port1.1, Port1.2 and so on

### CODE REVIEW

**Comment by sdszhang** [comment]
@amitpawar12 to review.

**Comment by sdszhang** [comment]
@selldinesh can you fix the pre-commit issue?

File: tests/common/snappi_tests/snappi_helpers.py [hunk 1]
```diff
@@ -182,17 +182,29 @@ def get_ports(self, peer_device=None):
...
             info_list = port.split('/')
-            dict_element = {
-                'ip': info_list[0],
-                'card_id': info_list[1].replace('Card', ''),
-                'port_id': info_list[2].replace('Port', ''),
-                'peer_port': info_list[3],
-                'peer_device': info_list[4],
-                'speed': info_list[5]
-            }
+            if 'Card' in port:
```
Comment (abdosi) [inline tests/common/snappi_tests/snappi_helpers.py:185]: @sdszhang: please not this change . After this PR we need make sure in links.csv we have to define line this `Card1/Port9`

File: tests/common/snappi_tests/snappi_helpers.py [hunk 2]
```diff
@@ -182,17 +182,29 @@ def get_ports(self, peer_device=None):
         retval = []
         for port in self.current_snappi_port_list:
             info_list = port.split('/')
-            dict_element = {
```
Comment (sdszhang) [inline tests/common/snappi_tests/snappi_helpers.py:185]: as @amitpawar12 suggested, please add some comment to show example for breakout definition, and sample definition that can be used for _link.csv.

**Comment by sdszhang** [comment]
@selldinesh can you fix the pre-commit error and add a comment for the breakout example?

File: tests/common/snappi_tests/snappi_helpers.py (hunk 2 above)
Comment (sdszhang) [reply tests/common/snappi_tests/snappi_helpers.py:185]: @selldinesh can you update the comment to have an example for breakout?

Comment (selldinesh) [reply tests/common/snappi_tests/snappi_helpers.py:185]: done

**Review by sdszhang — APPROVED** [review]

**Review by yejianquan — APPROVED** [review]
LGTM

**Pull request merged**